pkg/search/postgres/store.go

```diff
@@ -535,7 +535,7 @@ func (s *genericStore[T, PT]) deleteMany(ctx context.Context, identifiers []stri
 		q := search.NewQueryBuilder().AddDocIDs(identifierBatch...).ProtoQuery()
 
 		if err := RunDeleteRequestForSchema(ctx, s.schema, q, s.db); err != nil {
-			if !continueOnError {
+			if !continueOnError || errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
This is not exactly what we want.  If we return an err the caller will rollback the transaction.  If `continueOnError` is set we want the caller to never rollback and always commit.  That is a choice.  

So I think this line should remain `if !continueOnError {`. Then after we log the error on 541 we add in
```
if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
   break
}
```
That way the call was successful as far as the caller is concerned which is the point of `continueOnError`.